tests/api/compiler.cpp

```diff
@@ -212,15 +212,160 @@ TEST_F(WithCommandQueue, ModuleScopeConstantData) {
     EXPECT_EQ(result[5], 0);
 }
 
+TEST_F(WithCommandQueue, ProgramBinaryCompile) {
+    static const char* source = R"(
+      kernel void test(global uint *output) {
+        uint gid = get_global_id(0);
+        output[gid] = gid;
+      }
+    )";
+    auto program = CreateProgram(source);
+    const size_t gws = 4;
+    const size_t buffer_size = gws * sizeof(cl_uint);
+    auto buffer = CreateBuffer(CL_MEM_WRITE_ONLY, buffer_size);
+    cl_uint result[gws] = {0};
+    ASSERT_EQ(buffer_size, sizeof(result));
+
+    CompileProgram(program);
+    ASSERT_EQ(GetProgramBinaryType(program),
+              CL_PROGRAM_BINARY_TYPE_COMPILED_OBJECT);
+    auto compiled_binary = GetProgramBinary(program);
+    auto binary_program = CreateProgramWithBinary(compiled_binary);
+    ASSERT_EQ(GetProgramBinaryType(binary_program),
+              CL_PROGRAM_BINARY_TYPE_COMPILED_OBJECT);
+
```

Comment:
We could check that the kernel creation is correctly rejected prior to calling clBuildProgram.